src/wp-includes/webfonts-api/providers/class-wp-webfonts-google-provider.php

```diff
@@ -0,0 +1,184 @@
+					} else {
+						$normal_weights = array_merge( $normal_weights, $this->get_font_weights( $font['font-weight'] ) );
+					}
+				}
+
+				$normal_weights = array_unique( $normal_weights );
+				$italic_weights = array_unique( $italic_weights );
+
+				if ( empty( $italic_weights ) && ! empty( $normal_weights ) ) {
+					$url_part .= ':wght@' . implode( ';', $normal_weights );
+				} elseif ( ! empty( $italic_weights ) && empty( $normal_weights ) ) {
+					$url_part .= ':ital,wght@1,' . implode( ';', $normal_weights );
+				} elseif ( ! empty( $italic_weights ) && ! empty( $normal_weights ) ) {
+					$url_part .= ':ital,wght@0,' . implode( ';0,', $normal_weights ) . ';1,' . implode( ';1,', $italic_weights );
+				}
+
+				$font_display_url_parts[] = $url_part;
+			}
+
+			$font_families_urls[] = $this->root_url . '?family=' . implode( '&family=', $font_display_url_parts ) . '&display=' . $font_display;
+		}
+
+		return $font_families_urls;
+	}
+
+	/**
+	 * Organizes the webfonts by font-display.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @return array Sorted by font-display.
+	 */
+	private function organize_webfonts() {
+		$font_display_groups = array();
+
+		/*
+		 * Group by font-display.
+		 * Each font-display will need to be a separate request.
+		 */
+		foreach ( $this->webfonts as $webfont ) {
+			if ( ! isset( $webfont['font-display'] ) ) {
+				$webfont['font-display'] = 'fallback';
+			}
+
+			if ( ! isset( $font_display_groups[ $webfont['font-display'] ] ) ) {
+				$font_display_groups[ $webfont['font-display'] ] = array();
+			}
+			$font_display_groups[ $webfont['font-display'] ][] = $webfont;
+		}
+
+		/*
+		 * Iterate over each font-display group and group by font-family.
+		 * Multiple font-families can be combined in the same request,
+		 * but their params need to be grouped.
+		 */
+		foreach ( $font_display_groups as $font_display => $font_display_group ) {
+			$font_families = array();
+
+			foreach ( $font_display_group as $webfont ) {
+				if ( ! isset( $font_families[ $webfont['font-family'] ] ) ) {
+					$font_families[ $webfont['font-family'] ] = array();
+				}
+				$font_families[ $webfont['font-family'] ][] = $webfont;
+			}
+
+			$font_display_groups[ $font_display ] = $font_families;
+		}
+
+		return $font_display_groups;
+	}
+
+	/**
+	 * Get an array of font-weights from the given font-weights if using a space delimited string.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string $font_weights The font-weights string.
+	 *
+	 * @return array The font-weights array.
+	 */
+	protected function get_font_weights( $font_weights ) {
+		if ( false === strpos( $font_weights, ' ' ) ) {
+			return (array) $font_weights;
+		}
+
+		$font_weights = explode( ' ', $font_weights );
+
+		// If there are 2 values, treat them as a range.
+		if ( 2 === count( $font_weights ) ) {
+			$font_weights = range( $font_weights[0], $font_weights[1], 100 );
```

Comment:
We can cast to integers to make sure.

```suggestion
			$font_weights = range( (int) $font_weights[0], (int) $font_weights[1], 100 );
```

Reply:
Done. Thank you @mitogh!

Reply:
Thanks, fantastic work on this feature love it :shipit: